tests/test_odilo.py

```diff
@@ -0,0 +1,140 @@
+# encoding: utf-8
+from nose.tools import (
+    eq_, ok_
+)
+
+import os
+import json
+
+from odilo import (
+    MockOdiloAPI,
+    OdiloRepresentationExtractor,
+    OdiloBibliographicCoverageProvider
+)
+
+from model import (
+    Contributor,
+    DeliveryMechanism,
+    Edition,
+    Identifier,
+    Representation,
+    Subject,
+    Hyperlink,
+)
+
+from testing import DatabaseTest
+
+
+class OdiloTest(DatabaseTest):
+    def setup(self):
+        super(OdiloTest, self).setup()
+        self.collection = MockOdiloAPI.mock_collection(self._db)
+        base_path = os.path.split(__file__)[0]
+        self.resource_path = os.path.join(base_path, "files", "odilo")
+
+    def sample_json(self, filename):
+        path = os.path.join(self.resource_path, filename)
+        data = open(path).read()
+        return data, json.loads(data)
+
+
+class OdiloTestWithAPI(OdiloTest):
+    """Automatically create a MockOdiloAPI class during setup.
+    """
+
+    def setup(self):
+        super(OdiloTestWithAPI, self).setup()
+        self.api = MockOdiloAPI(self._db, self.collection)
+
+
+class TestOdiloBibliographicCoverageProvider(OdiloTest):
+    def setup(self):
+        super(TestOdiloBibliographicCoverageProvider, self).setup()
+        self.provider = OdiloBibliographicCoverageProvider(
+            self.collection, api_class=MockOdiloAPI
+        )
+        self.api = self.provider.api
+
+    def test_process_item(self):
+        print 'Testing process item...'
+        record_metadata, record_metadata_json = self.sample_json("odilo_metadata.json")
+        self.api.queue_response(200, content=record_metadata_json)
+        availability, availability_json = self.sample_json("odilo_availability.json")
+        self.api.queue_response(200, content=availability)
+
+        identifier, made_new = self.provider.process_item('00010982')
+        ok_(identifier, msg="Problem testing process item !!!")
```

Comment:
You should also test that `process_item` did what it's supposed to do -- that metadata and availability information were imported properly and that `handle_success` was called -- as well as verifying that the Identifier returned has the right `.type` and `.identifier`.

Reply:
Ok, I have revised the tests following your advice.

Reply:
Also I have added new tests for the API as you request below.